### PR TITLE
Fix for not added attribute, while added it dynamically from product form

### PR DESCRIPTION
**Pull request by bartoszkubicki**
### Description

PR fixes bug in product form, when adding attribute to product form.
Refactor of whole controller class, cleaning up dependencies.

### Manual testing scenarios

1. Open product form
2. Click add attribute (list seems to list all attributes not assigned to current attribute set)
3. Choose one attribute from form
4. Click add selected -> with fix applied attribute will appear in new group 'Attributes'. If attribute is not assigned to any attribute it will apear in product form (that works). If we choose attribute already assigned to attribute set it won't appear.

[issue-18206.zip](https://github.com/magento/magento2/files/2613375/issue-18206.zip)

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-engcom-team** [comment]
Hi @bartoszkubicki. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me $VERSION instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

**Comment by rogyar** [comment]
Hi @bartoszkubicki. Thank you for your contribution. As far as I see, we have a couple of code style adjustments in a scope of this PR. Could you point me to the exact place in the code which should fix the issue, please?
Also, consider fixing the failing tests. Thank you

**Comment by bartoszkubicki** [comment]
@rogyar I will fix the tests. Removing content of line 190 fixes the problem.

**Comment by bartoszkubicki** [comment]
@rogyar I could not locally reproduce these errros. Actually I don't understand what kind of short description I should put in requested lines:
`200 | ERROR | [x] Missing short description
 201 | ERROR | [x] There must be exactly one blank line before tags`

so I should modify phpDoc of function 'getAttributeteRepository()' by adding something like `Get attribute repository`? For me it is spare comment.

**Comment by rogyar** [comment]
Hi @bartoszkubicki. You can enable PHP Mess Detector and Code Sniffer in your IDE for highlighting code style issues. You can google for "Enable code sniffer and mess detector in Magento" for detailed instructions (I cannot provide you a direct link for my favourite resource here). 
As for the comment for getters, I agree. Sometimes they might be redundant but it's a new requirement that all methods should content comment in the DocBlock. It's necessary for the automated documentation generation and other purposes. 
Thank you.

**Comment by bartoszkubicki** [comment]
@rogyar I have pushed changes. I use CS but the I have it installed globally and I check code against PSR-2 standard. These short descriptions for me are spare, but it is not the point of this MR, so I have fixed errors. Hope, that now changes can be checked.

File: app/code/Magento/Catalog/Controller/Adminhtml/Product/AddAttributeToTemplate.php [hunk 1]
```diff
@@ -75,33 +83,33 @@ class AddAttributeToTemplate extends \Magento\Catalog\Controller\Adminhtml\Produ
...
      */
     public function __construct(
-        \Magento\Backend\App\Action\Context $context,
-        \Magento\Catalog\Controller\Adminhtml\Product\Builder $productBuilder,
-        \Magento\Framework\Controller\Result\JsonFactory $resultJsonFactory,
-        \Magento\Eav\Api\Data\AttributeGroupInterfaceFactory $attributeGroupFactory = null
+        Context $context,
+        Builder $productBuilder,
+        JsonFactory $resultJsonFactory,
+        AttributeGroupInterfaceFactory $attributeGroupFactory = null
```
Comment (sivaschenko) [inline app/code/Magento/Catalog/Controller/Adminhtml/Product/AddAttributeToTemplate.php:97]: The $attributeGroupFactory is longer than 20 characters, so it causes static tests to fail, however, we cannot change the variable name without bracking backward compatibility. In this exceptional situation there is nothing we can do, except adding a ` @SuppressWarnings(PHPMD.LongVariable)`

**Review by sivaschenko — CHANGES_REQUESTED** [review]

**Comment by sivaschenko** [comment]
Hi @bartoszkubicki , thanks for your contribution! Please see my review note.

File: app/code/Magento/Catalog/Controller/Adminhtml/Product/AddAttributeToTemplate.php (hunk 1 above)
Comment (bartoszkubicki) [reply app/code/Magento/Catalog/Controller/Adminhtml/Product/AddAttributeToTemplate.php:97]: Done, I will try to squash commits.

**Comment by bartoszkubicki** [comment]
@sivaschenko Done, you can check it again.

**Review by sivaschenko — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @sivaschenko, thank you for the review.
[ENGCOM-3519](https://jira.corp.magento.com/browse/ENGCOM-3519) has been created to process this Pull Request

**Comment by sivaschenko** [comment]
@bartoszkubicki we cannot reproduce the issue described in the PR on vanilla Magento, can you please provide more details for testing

**Comment by bartoszkubicki** [comment]
@sivaschenko I have corrected issue description and added video of what is going on and what exactly is buggy.

**Comment by magento-engcom-team** [comment]
Hi @bartoszkubicki. Thank you for your contribution. 
We will aim to release these changes as part of 2.3.1. 
Please check the release notes for final confirmation.

**Pull request merged**